app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt

```diff
@@ -0,0 +1,127 @@
+import com.example.itbook.base.BaseFragment
+import com.example.itbook.data.model.Book
+import com.example.itbook.data.repository.BooksRepository
+import com.example.itbook.data.source.local.BooksLocalDataSource
+import com.example.itbook.data.source.local.Database.BooksDatabase
+import com.example.itbook.data.source.local.dao.BookDaoImp
+import com.example.itbook.data.source.remote.BookRemoteHandler
+import com.example.itbook.data.source.remote.BooksRemoteDataSource
+import com.example.itbook.ui.adapter.PreviewBookAdapter
+import com.example.itbook.ui.detailbook.DetailBookFragment
+import com.example.itbook.ui.dialog.LoadingDialogFragment
+import com.example.itbook.utils.closeKeyboard
+import kotlinx.android.synthetic.main.fragment_favorite.*
+
+class FavoriteBooksFragment : BaseFragment(), FavoriteBooksContract.View {
+    override val layoutResource = R.layout.fragment_favorite
+
+    private var presenter: FavoriteBooksPresenter? = null
+    private val previewBookAdapter = PreviewBookAdapter(this::onBookClick)
+    private var loadingDialogFragment: LoadingDialogFragment? = null
+    private var books = mutableListOf<Book>()
+
+    override fun initViews() {
+        recyclerFavoriteBooks.adapter = previewBookAdapter
+        loadingDialogFragment = LoadingDialogFragment()
+        loadingDialogFragment?.isCancelable = false
+        isSetup = false
+    }
+
+    override fun initData() {
+
+        activity?.let {
+            val booksRepository = BooksRepository.getInstance(
+                BooksRemoteDataSource.getInstance(BookRemoteHandler()),
+                BooksLocalDataSource.getInstance(BookDaoImp(BooksDatabase.getInstance(it)))
+            )
+            presenter = FavoriteBooksPresenter(this, booksRepository)
+            presenter?.start()
+        }
+    }
+
+    override fun initListeners() {
+        editSearchFavoriteBooks.setOnEditorActionListener { _, _, _ ->
+            activity?.closeKeyboard(editSearchFavoriteBooks)
+            true
+        }
+        editSearchFavoriteBooks.addTextChangedListener(textWatcher)
+    }
+
+    override fun showFavoriteBooks(books: List<Book>) {
+        val booksCount = "${books.size} ${textBooksCount.text}"
+
+        textBooksCount.text = booksCount
+        previewBookAdapter.apply {
+            updateData(books.toMutableList())
+            notifyDataSetChanged()
+        }
+        this.books = books.toMutableList()
+    }
+
+    override fun showError(error: Exception?) {
+        Toast.makeText(activity, error.toString(), Toast.LENGTH_LONG).show()
+    }
+
+    override fun showLoading(isShow: Boolean) {
+        loadingDialogFragment?.let {
+            if (isShow) {
+                activity?.let { it1 -> it.show(it1.supportFragmentManager, "") }
+            } else {
+                it.dismiss()
+            }
+        }
+    }
+
+    private fun onBookClick(position: Int) {
+        val fragment = DetailBookFragment()
+        DetailBookFragment.apply {
```

Comment:
sao k applly từ trên mà còn thêm dòng lm j
```suggestion
        val fragment = DetailBookFragment().apply {
```